src/services/moderators.service.ts

```diff
@@ -0,0 +1,159 @@
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import * as bcrypt from 'bcrypt';
+import * as jwt from 'jsonwebtoken';
+import { Model } from 'mongoose';
+
+import { Moderator } from '../interfaces/moderator.interface';
+import { Post } from '../interfaces/post.interface';
+
+@Injectable()
+export class ModeratorsService {
+  PER_PAGE_ITEMS = 2;
+
+  constructor(
+    @InjectModel('Moderator') private readonly moderatorModel: Model<Moderator>,
+    @InjectModel('Post') private readonly postModel: Model<Post>,
+  ) {}
+
+  /**
+   * Moderator signup. Returns a success object or an error object.
+   * @param {string} firstName - moderator first name.
+   * @param {string} lastName - moderator last name.
+   * @param {string} username - username / handle.
+   * @param {string} email - moderator email.
+   * @param {string} password - moderator password.
+   */
+  async signup(
+    firstName: string,
+    lastName: string,
+    username: string,
+    email: string,
+    password: string,
+  ): Promise<any> {
```

Comment:
Using any type here nullifies the use of TypeScript, we can define an interface for it as well.